middlewares/validators/validator.helper.js

```diff
@@ -147,66 +124,30 @@ function nameValidator(fieldLocation, fieldname, optional = true) {
     }
 }
 
-// untested
 /**
- * Validates that field pronoun is ascii only.
- * @param {"query" | "body" | "header" | "param"} fieldLocation the location where the field should be found 
- * @param {string} fieldname name of the field that needs to be validated.
- * @param {boolean} optional whether the field is optional or not.
- */
-function pronounValidator(fieldLocation, fieldname, optional = true) {
-    const pronoun = setProperValidationChainBuilder(fieldLocation, fieldname, "invalid pronoun");
-    if (optional) {
-        return pronoun.optional({
-            checkFalsy: true
-        }).isAscii().withMessage("must contain only ascii characters");
-    } else {
-        return pronoun.exists().withMessage("pronoun must exist").isAscii().withMessage("must contain only ascii characters");
-    }
-}
-
-/**
- * Validates that field is a valid URL
+ * Validates the field against a regex
  * @param {"query" | "body" | "header" | "param"} fieldLocation the location where the field should be found 
  * @param {string} fieldname Name of the field that needs to be validated.
  * @param {boolean} optional Whether the field is optional or not.
- * @description 
- * Matches against a regex that looks for optional http://, https://, http:, https:, and optional www.
- * Regex then looks for the domain, and then optional route, path, query parameters
+ * @param {regex} desire The regex to match against
+ * @description The default regex to match against accepts anything
  */
-function urlValidator(fieldLocation, fieldname, optional = true) {
-    const url = setProperValidationChainBuilder(fieldLocation, fieldname, "invalid name");
+function regexValidator(fieldLocation, fieldname, optional = true, desire = Constants.ANY_REGEX) {
```

Comment:
Should there be a default `desire` value?

Reply:
My question is: is a default desire value necessary in this case?

Reply:
Ooooh I see what you mean. I'm not sure it's necessary, and it could be more transparent to force a regex argument. However, I was thinking of keeping the validator similar to the booleanValidator, where the desire is optional, and the default value is the most encompassing. What do you think?

Reply:
Up to you!

Reply:
Going to leave as is.